src/app/core/config/config.service.ts

```diff
@@ -48,4 +49,88 @@ export class ConfigService extends LatestEntityLoader<Config> {
     }
     return matchingConfigs;
   }
+
+  private applyMigrations(config: Config): Config {
+    const migrations: ConfigMigration[] = [
+      migrateFormHeadersIntoFieldGroups,
+      migrateFormFieldConfigView2ViewComponent,
+    ];
+
+    const newConfig = JSON.parse(JSON.stringify(config), (_that, rawValue) => {
+      let configPart = rawValue;
+      for (const migration of migrations) {
+        if (migration.filter(_that, configPart)) {
```

Comment:
The word `filter` is confusing me here a little as its not really filtering anything. What about `shouldMigrate`?

Reply:
true - we could actually also leave this within the single migration function and let them just return the object unchanged if nothing should be migrated.

I would suggest to make `type ConfigMigration = (key: string, configPart: any) => any`, i.e. a "reviver" function implementation. And then we simply chain all migration revivers within the JSON.parse to apply migrations.